### DISCUSSION TITLE
Flip Taking

### DISCUSSION THREAD

**Discussion by simontakouridis** [idena-network/idena-docs#99] [IIPs]
<h1><font size="4" style="font-size: 16pt"><b>IIP: Flip Taking</b></font></h1>

<h2 style="margin-bottom: 0cm; line-height: 100%"><font size="4" style="font-size: 14pt"><b>Abstract</b></font></h2>

<p style="margin-bottom: 0cm; line-height: 100%">This proposal
introduces the functionality of taking (un-submitted) flips from
other users. This in effect makes flip submission optional for users
who do not wish to participate, yet still wish to benefit from all
other aspects of Idena. It also enables those that enjoy making quality
flips to benefit from their efforts.</p>

<h2 style="margin-bottom: 0cm; line-height: 100%"><font size="4" style="font-size: 14pt"><b>Motivation</b></font></h2>

<p style="margin-bottom: 0cm; line-height: 100%">A growing base of
genuine users is needed for any project in its infancy. In fact, many
businesses fail not because the idea was not world changing, but
because the idea failed to gain traction. Idena is vulnerable with
its current low market capitalisation for several reasons, but most
notably the challenge to pay for development and operational costs,
and furthermore potential investor may opt to fork the protocol any
create their own coin, rather than buy into Idena. As such,
significant traction is expected to provide the momentum to allay
these threats, and moreover reach its potential for mass user
adoption.</p>

<p style="margin-bottom: 0cm; line-height: 100%">Flip quality is an
essential component of Idena's security layer, which has been declining
over time to the point that the network may be vulnerable to AI
attack.</p>

<p style="margin-bottom: 0cm; line-height: 100%">Here I introduce a
significant protocol change to make flip creation optional (with
conditions), thereby increasing the scalability and popularity of the
network. Anyone with a remote interest in Idena can now participate
without significant investment of time and energy, thus lowering the
barrier to entry and maintenance. With this change Idena will be
inclusive of all people, not just to those with a hardcore commitment
to the project. This change also enables specialisation to occur with
regards to flip making, for better overall flip quality and thus
enahnced security.</p>

<h2 style="margin-bottom: 0cm; line-height: 100%"><font size="4" style="font-size: 14pt"><b>Specification</b></font></h2>

<p style="margin-bottom: 0cm; line-height: 100%"><h3>Overview</b></h3>

<p style="margin-bottom: 0cm; line-height: 100%">In the beginning of
each epoch, a validator has a quota of 3 required flips in order to
be validated for the next epoch. This requirement is maintained with
this proposal, however the quota of un-submitted flips can be reduced
by other validated identities “taking” them. More specifically,
when a validator has submitted all of their own required flips, they
are then provided with the option to take an additional flip from
someone else’s required flip quota. When selecting this option, the
protocol will find a random validator with remaining quota and
essentially take it. Over epoch time, this process results in all
flip creation being completely taken from those validators that
either opted to not submit any flips, or placed no effort into
submitting their flips in a timely manner.</p>

<p style="margin-bottom: 0cm; line-height: 100%">There are certain
rules surrounding this process, such as a maximum number of flips
that can be taken by any particular validator, the rate at which they
can be taken, and the priority of which validators have their flips
taken from them. Furthermore, changes to the protocol incentives and
rules are updated to accommodate this new functionality.</p>

<p style="margin-bottom: 0cm; line-height: 100%"><h3>Change in
Validation Rewards</b></h3>

<p style="margin-bottom: 0cm; line-height: 100%">Current random
sampling indicates that a very low number of validators utilise their
“optional” flips (~10% of validators), meaning current flip
rewards are insufficient to motivate validators to do any more than
what is required. As such, greater flip rewards are needed.</p>

<p style="margin-bottom: 0cm; line-height: 100%">The flip rewards
will be raised to 51% of validation rewards. Moreover, flip reporting
rewards will be raised from 15% to 45% to incentivize good flip reporting.
These reward changes will be accomodated by removal of the following
validation rewards:</p>

<ul>
	<li><p style="margin-bottom: 0cm; line-height: 100%">Staking rewards
	(18%). These rewards are redundant now that staked mining rewards
	have been implemented.</p>
	<li><p align="left" style="margin-bottom: 0cm; line-height: 100%">Invitation
	rewards (18%). These rewards are an absurdity, people are motivated
	to invite family/friends for reasons other than receiving a
	transactional reward.</p>
	<li><p align="left" style="margin-bottom: 0cm; line-height: 100%">Idena
	foundation payouts (10%). These rewards are redundant because the
	Foundation Wallet has already accumulated ~1,366,000 iDNA, and the
	Zero Wallet continues to accumulate iDNA.</p>
</ul>

</p>

<h3 style="margin-bottom: 0cm; line-height: 100%"><b>Removal of
Optional Flips</b></h3>

<p style="margin-bottom: 0cm; line-height: 100%">Optional flips (as
currently implemented) for both verified and human status, are now
inconsistent with this proposal – flips are either required or
taken.</p>

<h3 style="margin-bottom: 0cm; line-height: 100%"><b>Taking Priority
Schedule</b></h3>

<p style="margin-bottom: 0cm; line-height: 100%">Available flips are
taken from other validators via the Taking Priority Schedule (See
Table 1). More specifically, available flips are taken from the first
priority level until no more are available, then taken from the
second priority level, and so forth. Within each priority level,
available flips are taken randomly. Validators may elect for their
flips to be prioritised for taking, resulting in that validator
rising in priority level. By default, election is turned on and
validators must turn off to de-prioritise their flips.</p>

<p style="margin-top: 0.21cm; margin-bottom: 0.21cm; line-height: 100%; page-break-after: avoid">
<font size="3" style="font-size: 12pt"><i>Table <span style="background: #c0c0c0">1</span>:
Taking Priority Schedule.</i></font></p>
<table width="193" cellpadding="0" cellspacing="0">
	<tr valign="top">
		<td width="46" style="border: none; padding: 0cm"><p><b>Priority
			Level</b></p>
		</td>
		<td width="57" style="border: none; padding: 0cm"><p><b>Status</b></p>
		</td>
		<td width="39" style="border: none; padding: 0cm"><p><b>Age</b></p>
		</td>
		<td width="51" style="border: none; padding: 0cm"><p><b>Elected</b></p>
		</td>
	</tr>
	<tr valign="top">
		<td width="46" style="border: none; padding: 0cm"><p>1</p>
		</td>
		<td width="57" style="border: none; padding: 0cm"><p>Newbie</p>
		</td>
		<td width="39" style="border: none; padding: 0cm"><p>1</p>
		</td>
		<td width="51" style="border: none; padding: 0cm"><p>Yes</p>
		</td>
	</tr>
	<tr valign="top">
		<td width="46" style="border: none; padding: 0cm"><p>2</p>
		</td>
		<td width="57" style="border: none; padding: 0cm"><p>Verified</p>
		</td>
		<td width="39" style="border: none; padding: 0cm"><p>2</p>
		</td>
		<td width="51" style="border: none; padding: 0cm"><p>Yes</p>
		</td>
	</tr>
	<tr valign="top">
		<td width="46" style="border: none; padding: 0cm"><p>3</p>
		</td>
		<td width="57" style="border: none; padding: 0cm"><p>Verified</p>
		</td>
		<td width="39" style="border: none; padding: 0cm"><p>3</p>
		</td>
		<td width="51" style="border: none; padding: 0cm"><p>Yes</p>
		</td>
	</tr>
	<tr valign="top">
		<td width="46" style="border: none; padding: 0cm"><p>4</p>
		</td>
		<td width="57" style="border: none; padding: 0cm"><p>Human</p>
		</td>
		<td width="39" style="border: none; padding: 0cm"><p>4</p>
		</td>
		<td width="51" style="border: none; padding: 0cm"><p>Yes</p>
		</td>
	</tr>
	<tr valign="top">
		<td width="46" style="border: none; padding: 0cm"><p>5</p>
		</td>
		<td width="57" style="border: none; padding: 0cm"><p>Human</p>
		</td>
		<td width="39" style="border: none; padding: 0cm"><p>5</p>
		</td>
		<td width="51" style="border: none; padding: 0cm"><p>Yes</p>
		</td>
	</tr>
	<tr valign="top">
		<td width="46" style="border: none; padding: 0cm"><p>6</p>
		</td>
		<td width="57" style="border: none; padding: 0cm"><p>Human</p>
		</td>
		<td width="39" style="border: none; padding: 0cm"><p>6</p>
		</td>
		<td width="51" style="border: none; padding: 0cm"><p>Yes</p>
		</td>
	</tr>
	<tr valign="top">
		<td width="46" style="border: none; padding: 0cm"><p>7</p>
		</td>
		<td width="57" style="border: none; padding: 0cm"><p>Human</p>
		</td>
		<td width="39" style="border: none; padding: 0cm"><p>7</p>
		</td>
		<td width="51" style="border: none; padding: 0cm"><p>Yes</p>
		</td>
	</tr>
	<tr valign="top">
		<td width="46" style="border: none; padding: 0cm"><p>8</p>
		</td>
		<td width="57" style="border: none; padding: 0cm"><p>Human</p>
		</td>
		<td width="39" style="border: none; padding: 0cm"><p>8</p>
		</td>
		<td width="51" style="border: none; padding: 0cm"><p>Yes</p>
		</td>
	</tr>
	<tr valign="top">
		<td width="46" style="border: none; padding: 0cm"><p>9</p>
		</td>
		<td width="57" style="border: none; padding: 0cm"><p>Human</p>
		</td>
		<td width="39" style="border: none; padding: 0cm"><p>9</p>
		</td>
		<td width="51" style="border: none; padding: 0cm"><p>Yes</p>
		</td>
	</tr>
	<tr valign="top">
		<td width="46" style="border: none; padding: 0cm"><p>10</p>
		</td>
		<td width="57" style="border: none; padding: 0cm"><p>Human</p>
		</td>
		<td width="39" style="border: none; padding: 0cm"><p>10+</p>
		</td>
		<td width="51" style="border: none; padding: 0cm"><p>Yes</p>
		</td>
	</tr>
	<tr valign="top">
		<td width="46" style="border: none; padding: 0cm"><p>11</p>
		</td>
		<td width="57" style="border: none; padding: 0cm"><p>Newbie</p>
		</td>
		<td width="39" style="border: none; padding: 0cm"><p>1</p>
		</td>
		<td width="51" style="border: none; padding: 0cm"><p>No</p>
		</td>
	</tr>
	<tr valign="top">
		<td width="46" style="border: none; padding: 0cm"><p>12</p>
		</td>
		<td width="57" style="border: none; padding: 0cm"><p>Verified</p>
		</td>
		<td width="39" style="border: none; padding: 0cm"><p>2</p>
		</td>
		<td width="51" style="border: none; padding: 0cm"><p>No</p>
		</td>
	</tr>
	<tr valign="top">
		<td width="46" style="border: none; padding: 0cm"><p>13</p>
		</td>
		<td width="57" style="border: none; padding: 0cm"><p>Verified</p>
		</td>
		<td width="39" style="border: none; padding: 0cm"><p>3</p>
		</td>
		<td width="51" style="border: none; padding: 0cm"><p>No</p>
		</td>
	</tr>
	<tr valign="top">
		<td width="46" style="border: none; padding: 0cm"><p>14</p>
		</td>
		<td width="57" style="border: none; padding: 0cm"><p>Human</p>
		</td>
		<td width="39" style="border: none; padding: 0cm"><p>4</p>
		</td>
		<td width="51" style="border: none; padding: 0cm"><p>No</p>
		</td>
	</tr>
	<tr valign="top">
		<td width="46" style="border: none; padding: 0cm"><p>15</p>
		</td>
		<td width="57" style="border: none; padding: 0cm"><p>Human</p>
		</td>
		<td width="39" style="border: none; padding: 0cm"><p>5</p>
		</td>
		<td width="51" style="border: none; padding: 0cm"><p>No</p>
		</td>
	</tr>
	<tr valign="top">
		<td width="46" style="border: none; padding: 0cm"><p>16</p>
		</td>
		<td width="57" style="border: none; padding: 0cm"><p>Human</p>
		</td>
		<td width="39" style="border: none; padding: 0cm"><p>6</p>
		</td>
		<td width="51" style="border: none; padding: 0cm"><p>No</p>
		</td>
	</tr>
	<tr valign="top">
		<td width="46" style="border: none; padding: 0cm"><p>17</p>
		</td>
		<td width="57" style="border: none; padding: 0cm"><p>Human</p>
		</td>
		<td width="39" style="border: none; padding: 0cm"><p>7</p>
		</td>
		<td width="51" style="border: none; padding: 0cm"><p>No</p>
		</td>
	</tr>
	<tr valign="top">
		<td width="46" style="border: none; padding: 0cm"><p>18</p>
		</td>
		<td width="57" style="border: none; padding: 0cm"><p>Human</p>
		</td>
		<td width="39" style="border: none; padding: 0cm"><p>8</p>
		</td>
		<td width="51" style="border: none; padding: 0cm"><p>No</p>
		</td>
	</tr>
	<tr valign="top">
		<td width="46" style="border: none; padding: 0cm"><p>19</p>
		</td>
		<td width="57" style="border: none; padding: 0cm"><p>Human</p>
		</td>
		<td width="39" style="border: none; padding: 0cm"><p>9</p>
		</td>
		<td width="51" style="border: none; padding: 0cm"><p>No</p>
		</td>
	</tr>
	<tr valign="top">
		<td width="46" style="border: none; padding: 0cm"><p>20</p>
		</td>
		<td width="57" style="border: none; padding: 0cm"><p>Human</p>
		</td>
		<td width="39" style="border: none; padding: 0cm"><p>10+</p>
		</td>
		<td width="51" style="border: none; padding: 0cm"><p>No</p>
		</td>
	</tr>
</table>

<h3 style="margin-bottom: 0cm; line-height: 100%"><b>Flip taking</b></h3>

<p style="margin-bottom: 0cm; line-height: 100%">When a non-newbie
validator has submitted all of their required flips, they are then
eligible to take a flip. When a flip is taken, a flip taking
transaction is created with reference of the event for storage on the
blockchain. The validator from which the flip was taken (the giver),
no longer has this required flip available to them. And the receiving
validator (the taker) now has an added required flip, which must be
submitted either prior the next validation, or prior to taking
another flip. Note that once a flip is taken, it may not be re-taken
by another validator - it must be submitted.</p>

<p style="margin-bottom: 0cm; line-height: 100%">The keywords used
for the taken flip are generated by the taker. By default, the 9
default keywords first generated by a validator occupy the first 9
determinant keywords (1..9), which are used with the first 3
submitted flips. For the next 3 flips submitted, i.e. for taken
flips, the validator will generate the next 9 determinant keywords
(10..18), and so on.</p>

<p style="margin-bottom: 0cm; line-height: 100%">Come the validation
ceremony, the flip lottery proceeds as per normal as if no flips been
taken from anyone, with one caveat which is a validator cannot be paired
with another validator for whom they have taken flips from. Blockchain
pointers (flip taking transactions) are utilised to locate the true flip
creator and flip, for gathering the images, keywords, and correctly
distributing rewards/penalties for that flip.</p>

<h3 style="margin-bottom: 0cm; line-height: 100%"><b>The maximum
number of flip takes</b></h3>

<p style="margin-bottom: 0cm; line-height: 100%">It takes time and
practice for a validator to become a skilled flip creator. As such,
limits will be placed for the maximum number of flips that a user can
create in order to ensure overall network flip quality, and also to
prevent abuse. More specifically, for each successful validation, a
validator will have their maximum flip submissions raised by 3 (See
Table 2). Moreover, a hard upper limit of 30 flips per validator is
designed to maintain decentralisation. Note that with this proposal,
validators can be penalised for unqualified/reported flips, which will
reduce their maximum flip submissions (more details below).</p>

<p style="margin-top: 0.21cm; margin-bottom: 0.21cm; line-height: 100%; page-break-after: avoid">
<font size="3" style="font-size: 12pt"><i>Table <span style="background: #c0c0c0">2</span>:
Typical validator maximum flip submissions.</i></font></p>
<table width="190" cellpadding="0" cellspacing="0">
	<tr valign="top">
		<td width="62" style="border: none; padding: 0cm"><p><b>Status</b></p>
		</td>
		<td width="46" style="border: none; padding: 0cm"><p><b>Age</b></p>
		</td>
		<td width="82" style="border: none; padding: 0cm"><p><b>Maximum
			flip submissions</b></p>
		</td>
	</tr>
	<tr valign="top">
		<td width="62" style="border: none; padding: 0cm"><p>Candidate</p>
		</td>
		<td width="46" style="border: none; padding: 0cm"><p>0</p>
		</td>
		<td width="82" style="border: none; padding: 0cm"><p>0</p>
		</td>
	</tr>
	<tr valign="top">
		<td width="62" style="border: none; padding: 0cm"><p>Newbie</p>
		</td>
		<td width="46" style="border: none; padding: 0cm"><p>1</p>
		</td>
		<td width="82" style="border: none; padding: 0cm"><p>3</p>
		</td>
	</tr>
	<tr valign="top">
		<td width="62" style="border: none; padding: 0cm"><p>Verified</p>
		</td>
		<td width="46" style="border: none; padding: 0cm"><p>2</p>
		</td>
		<td width="82" style="border: none; padding: 0cm"><p>6</p>
		</td>
	</tr>
	<tr valign="top">
		<td width="62" style="border: none; padding: 0cm"><p>Verified</p>
		</td>
		<td width="46" style="border: none; padding: 0cm"><p>3</p>
		</td>
		<td width="82" style="border: none; padding: 0cm"><p>9</p>
		</td>
	</tr>
	<tr valign="top">
		<td width="62" style="border: none; padding: 0cm"><p>Human</p>
		</td>
		<td width="46" style="border: none; padding: 0cm"><p>4</p>
		</td>
		<td width="82" style="border: none; padding: 0cm"><p>12</p>
		</td>
	</tr>
	<tr valign="top">
		<td width="62" style="border: none; padding: 0cm"><p>Human</p>
		</td>
		<td width="46" style="border: none; padding: 0cm"><p>5</p>
		</td>
		<td width="82" style="border: none; padding: 0cm"><p>15</p>
		</td>
	</tr>
	<tr valign="top">
		<td width="62" style="border: none; padding: 0cm"><p>Human</p>
		</td>
		<td width="46" style="border: none; padding: 0cm"><p>6</p>
		</td>
		<td width="82" style="border: none; padding: 0cm"><p>18</p>
		</td>
	</tr>
	<tr valign="top">
		<td width="62" style="border: none; padding: 0cm"><p>Human</p>
		</td>
		<td width="46" style="border: none; padding: 0cm"><p>7</p>
		</td>
		<td width="82" style="border: none; padding: 0cm"><p>21</p>
		</td>
	</tr>
	<tr valign="top">
		<td width="62" style="border: none; padding: 0cm"><p>Human</p>
		</td>
		<td width="46" style="border: none; padding: 0cm"><p>8</p>
		</td>
		<td width="82" style="border: none; padding: 0cm"><p>24</p>
		</td>
	</tr>
	<tr valign="top">
		<td width="62" style="border: none; padding: 0cm"><p>Human</p>
		</td>
		<td width="46" style="border: none; padding: 0cm"><p>9</p>
		</td>
		<td width="82" style="border: none; padding: 0cm"><p>27</p>
		</td>
	</tr>
	<tr valign="top">
		<td width="62" style="border: none; padding: 0cm"><p>Human</p>
		</td>
		<td width="46" style="border: none; padding: 0cm"><p>10+</p>
		</td>
		<td width="82" style="border: none; padding: 0cm"><p>30</p>
		</td>
	</tr>
</table>

<h3 style="margin-bottom: 0cm; line-height: 100%"><b>Unqualified/Reported flip
penalties</b></h3>

<p style="margin-bottom: 0cm; line-height: 100%">Because a validator
can create many more flips, they can do more harm to the network via
intentional (or unintentional) shitflips – as such, penalisations
are required. Specifically, a validators’ qualified flip creation
score for any given epoch should be 60% or more. A validators’
total qualified flip creation score for the last 10 epochs should be
75% or more. Further, a validator cannot have more than 2 reported
flips for the last 10 epochs. If any of these thresholds are unmet,
the validator loses their validated status and has their maximum flip
submissions reduced back to 0 (Candidate level).</p>

<h3 style="margin-bottom: 0cm; line-height: 100%"><b>Reduced flip
submission rate</b></h3>

<p style="margin-bottom: 0cm; line-height: 100%"><span style="font-weight: normal">The
f</span><span style="font-weight: normal">lip submission </span><span style="font-weight: normal">rate
</span><span style="font-weight: normal">per validator </span><span style="font-weight: normal">will
be set to a </span><span style="font-weight: normal">maximum </span><span style="font-weight: normal">of
</span><span style="font-weight: normal">3 </span><span style="font-weight: normal">flips</span><span style="font-weight: normal">
per hour. </span><span style="font-weight: normal">This is d</span>esigned
to provide all other validators with sufficient time to create their
own flips, and not having them taken prior to them having the chance
to submit them. It is also designed to prevent too many transactions
happening in one go at the beginning of the next epoch.</p>

<p style="margin-bottom: 0cm; line-height: 100%">Note that an
intentional design of this proposal is to disallow users from
preventing their flips from being taken. If users could prevent 
their flips from being taken and they subsequently fail to
submit their flips, this would decrease the number of flips in the
network and jeopardise the optionality of flip creation for everyone
else. If validator wish to guarantee that their flip is not taken,
then they should put effort into submitting early in the epoch.</p>

<h3 style="margin-bottom: 0cm; line-height: 100%"><b>Taking time limit</b></h3>

<p style="margin-bottom: 0cm; line-height: 100%">In order to provide
validators with certainty about whether or not they have required flips
to create and submit, flip taking is time limited to the first half of the
epoch. With this limitation, a validator that can now check half way
through the epoch and know for certain if they will be required to make
any flips. It also provides ample time for these required flips to be
created and submitted.</p>

<h2 style="margin-bottom: 0cm; line-height: 100%"><font size="4" style="font-size: 14pt"><b>Rationale</b></font></h2>

<p style="margin-bottom: 0cm; line-height: 100%">The amount of
individual time and effort required for Idena participation is
extraordinarily high, which is mostly attributable to mandatory flip
creation. Most people willing are either those that are super
passionate in the technology and potential, and/or stakers that wish
to profit from the system rewards. In either case, the high effort
involved with mandatory flip creation significantly denudes the
immense growth potential from people that are potentially interested
in the technology and/or rewards, but are unwilling to invest the
time and effort. Furthermore, the burden of flip creation is
currently placed on people who are simply uninterested.</p>

<p style="margin-bottom: 0cm; line-height: 100%">The proposal herein
provides aligned incentives for two new critical target groups of
users, making Idena an attractive value proposition expected to
result in immense growth of the network. The first group are users
who can earn significant rewards by creating flips, and are rewarded
directly proportional to the time and effort provided. The second
group and by far the largest, are ordinary users that have an
interest in Idena, and can now participate without the associated
burden.</p>

<p style="margin-bottom: 0cm; line-height: 100%">The key benefits of
this change is to improve the quality of flips produced by specialised workers,
and to expand Idena participation, which is expected to increase:</p>
<ul>
	<li><p style="margin-bottom: 0cm; line-height: 100%">conversion of
	participants into investors, miners, word of mouth promoters.</p>
	<li><p style="margin-bottom: 0cm; line-height: 100%">eyeballs for
	Idena advertisements.</p>
	<li><p style="margin-bottom: 0cm; line-height: 100%">interest from
	third-party applications for using Idena Sign-In, and Idena
	Advertising (if third-party platform revenues are enabled).</p>
	<li><p style="margin-bottom: 0cm; line-height: 100%">interest from
	Web3 and DAO developers (for when iwasm is enabled).</p>
</ul>

<p style="margin-bottom: 0cm; line-height: 100%">It also provides a
pathway for flip creators to earn iDNA for participation in staked
mining rewards.</p>

<h2 style="margin-bottom: 0cm; line-height: 100%"><font size="4" style="font-size: 14pt"><b>Backward
Compatibility</b></font></h2>

<p style="margin-bottom: 0cm; line-height: 100%">This change requires
a hard fork.</p>

<h2 style="margin-bottom: 0cm; line-height: 100%"><font size="4" style="font-size: 14pt"><b>Security
Considerations</b></font></h2>

<ul>
	<li><p style="margin-bottom: 0cm; line-height: 100%">Flip production
	will have increased concentration and can lead to power
	concentration. This is mitigated by introducing maximum flip
	submissions.</p>
	<li><p style="margin-bottom: 0cm; line-height: 100%">Competitiveness
	to take flips is introduced which can disrupt a good user
	experience. This is mitigated by introducing a reduced flip
	submission rate.</p>
	<li><p style="margin-bottom: 0cm; line-height: 100%">Inconsistent
	user requirements may arise if in some epochs flip submission was
	not required, yet in other epochs submission was required. To
	mitigate this, flip rewards have been significantly maximised so
	that users shouldn’t ever have to submit flips, and furthermore
	flips cannot be protected from taking, thus ensuring maximal flip
	production from those that want to submit.</p>
</ul>

**Comment by 0JustinMiles0** [comment]
My main feedback on this proposal is on the motivation/rational sections.

The proposal mentions:

"Anyone with a remote interest in Idena can now participate without significant investment of time and energy, thus lowering the barrier to entry and maintenance. With this change Idena will be inclusive of all people, not just to those with a hardcore commitment to the project. [...] The amount of individual time and effort required for Idena participation is extraordinarily high, which is mostly attributable to mandatory flip creation. Most people willing are either those that are super passionate in the technology and potential, and/or stakers that wish to profit from the system rewards. In either case, the high effort involved with mandatory flip creation significantly denudes the immense growth potential from people that are potentially interested in the technology and/or rewards, but are unwilling to invest the time and effort. Furthermore, the burden of flip creation is currently placed on people who are simply uninterested."

If I understand correctly, the proposal aims to:
1) attract more participants by 
2) making flips creation optional as
3) it is the most demanding and time consuming activity

I would challenge assumption number 3 as I don't think that flip creation is what people see as the most demanding and time consuming activity. In my opinion, flip creation is magnitude less demanding and time consuming than attending the ceremony at a specific date and time. Hence, I don't think that making flips creation optional would remove a significant burden for potential participants who decide to not be involved because they see their participation as demanding too much of their time and efforts. 

**In order to validate the assumption that flips creation would remove a significant burden to network participation, I would suggest to create an oracle polling current network participants on what they see as the most demanding and time consuming activities for their participation.**

**Reply by simontakouridis** [reply]
Thanks Justin for the feedback, I have updated the Motivation and Rationale sections to reflect the other benefit of this proposal, which is to introduce specialisation with regards to flip creation for better overall flip quality and thus enhanced security, in addition to the reduced burden on participation.

With regards to your other feedback, there have been many grievances expressed over time on Discord about the cumbersomeness of making flips, and the fact that optional flips are almost never utilised are sufficient evidence that the vast majority of validators do not want to do it, regardless of whether it is the most (or second most) demanding activity. Whether this translates into growth will need to be actually tested, and no oracle will be able to predict this.

**Reply by 0JustinMiles0** [reply]
Thanks for adding these elements. However, I'm not sure I understand how having specialized workers would necessarily increase the quality of the flips since there is no indicator for quality control on flips or flip makers. Is there a way this system would prevent low-quality flip makers from taking flips? If not this system may potentially produce counterproductive results as flip farms (which generally create lower quality flips) are good at mass producing flips. 

Regarding the assertion that flip making is demanding and time consuming for many participants, I certainly think that this is true to some extent but again I think this barrier to participation is significantly lower than attending a specific data and time. I still think that we could use Oracles to have a better read (Discord only represents a small subset of network participants) on the level of pain this proposal would solve. There are a few different questions we could ask. It could be as simple as asking participants how long it takes them in average to create 3 flips. If we see a large distribution between participants across various time frame then, that would indicate that the flips making activity is a lot more cumbersome for some participants than others which may validate the rational of this proposal.

This proposal certainly solves a pain but because it involves significant changes, I would want the pain to be better qualified in the first place. Then, risks and unintended consequences that this proposal could introduce can be further discuss.

**Reply by simontakouridis** [reply]
With regards to flip quality, this proposal focusses on experience of the flip creator, but does not prevent abuse - that would require a different iip which is out of the scope here. However, I think we could direct more of the generous flip rewards in this iip towards reporting rewards, this could definitely help. E.g. 51% flip rewards, 45% reporting rewards.

If I were to create an oracle, I'd prefer to ask whether the community supports this proposal. Or asking a question that helps to amend this proposal in a way that improves it. Asking whether flip creating is painful or not is nonsensicle to me, because it's obviously a burden. And it is not just current users that should be considered, it is also new users or even lost users. I cannot imagine a future where Idena is mainstream and everyone is creating flips. Specialisation and optionality needs to happen for mainstream adoption.

**Reply by 0JustinMiles0** [reply]
I certainly can see your point when it comes to mainstream adoption and I agree that not everybody would find it "fun" to create flips.

**Reply by simontakouridis** [reply]
I'm going to update the rewards distribution, 51% flip rewards, 45% reporting rewards. Thanks for your feedback Justin!

**Reply by 0JustinMiles0** [reply]
Of course!
One more thing on flip quality, although flip reporting is still very useful to prevent very bad flips from being created, it has reached its limits as there are many ways low-quality flips can still be submitted and approved. As you said, this isn't the goal of this proposal to address this issue but I'm somewhat concerned that, until we have a better system to incentivize flip makers to create high-quality flips, this proposal could open the door to more low-quality flips as you can be sure that flip farms will be among the first ones to try to maximize how many additional flips they can take for themselves.

**Reply by simontakouridis** [reply]
If I had it my way, flip quality would have been solved long ago. But yeah, I think I may agree with you to an extent that flip quality may need to be prioritised.

**Comment by simontakouridis** [comment]
Please note that the Specification was updated to include the "Taking time limit".

**Comment by simontakouridis** [comment]
Please note a couple more updates to the Specification: 1) once a flip is taken, it may not be re-taken by another validator - it must be submitted. 2) during the flip lottery, a validator cannot be paired with another validator for whom they have taken flips from.

**Comment by simontakouridis** [comment]
Please note another update, a validator will be penalised if having more than 2 reported flips for the last 10 epochs.
